src/constraints/load_balance.jl

```diff
@@ -43,21 +55,33 @@ function add_elec_load_balance_constraints(m, p; _n="")
 	##Constraint (8b): Electrical Load Balancing without Grid
 	if !p.s.settings.off_grid_flag # load balancing constraint for grid-connected runs
         @constraint(m, [ts in p.time_steps_without_grid],
-            sum(p.production_factor[t,ts] * p.levelization_factor[t] * m[Symbol("dvRatedProduction"*_n)][t,ts] for t in p.techs.elec)  
+            sum(p.production_factor[t, ts] * p.levelization_factor[t] * m[Symbol("dvRatedProduction"*_n)][t,ts]
+                - sum(m[Symbol("dvProductionToStorage"*_n)][b, t, ts] for b in p.s.storage.types.ac_coupled)
+                - m[Symbol("dvCurtail"*_n)][t, ts]
+                for t in p.techs.ac_coupled_with_storage)
+            + sum(p.s.storage.attr["ElectricStorage"].inverter_efficiency_fraction * ( # convert DC to AC
+                    p.production_factor[t, ts] * p.levelization_factor[t] * m[Symbol("dvRatedProduction"*_n)][t,ts]
+                    - sum(m[Symbol("dvProductionToStorage"*_n)][b, t, ts] for b in p.s.storage.types.dc_coupled)
+                    - m[Symbol("dvCurtail"*_n)][t, ts])
+                for t in p.techs.dc_coupled_with_storage)
             + sum(m[Symbol("dvDischargeFromStorage"*_n)][b,ts] for b in p.s.storage.types.elec)
             ==
-            sum(sum(m[Symbol("dvProductionToStorage"*_n)][b, t, ts] for b in p.s.storage.types.elec) 
-                + m[Symbol("dvCurtail"*_n)][t, ts] for t in p.techs.elec)
-            + p.s.electric_load.critical_loads_kw[ts]
+            p.s.electric_load.critical_loads_kw[ts]
         )
     else # load balancing constraint for off-grid runs 
         @constraint(m, [ts in p.time_steps_without_grid],
-            sum(p.production_factor[t,ts] * p.levelization_factor[t] * m[Symbol("dvRatedProduction"*_n)][t,ts] for t in p.techs.elec)
+            sum(p.production_factor[t, ts] * p.levelization_factor[t] * m[Symbol("dvRatedProduction"*_n)][t,ts]
```

Comment:
is this just partially set up for off-grid runs for if you have time to finish this in the future (with the OR constraints)?